### ISSUE TITLE
Error occurred

### ISSUE THREAD

**Issue by LiPan123456789** [bdwsq1996/Effi-MVS#3]
depth_loss = F.smooth_l1_loss(depth_est[mask], depth_gt[mask], reduction='mean')

The problem is in line 513 of the module under models. When I set the batchsize to 8, depth_est.shape[8,64,80],depth_gt.shape[8,8,10],mask.shape[8,8,10].

I guess it is because depth_est has not been resized to a list structure like stage1, stage2, stage3, stage4, so it reports an error. Please check the following to see if this is the problem, thanks!

**Comment by bdwsq1996** [comment]
I tested my code again, but I didn't find this problem. During the training, the size of the image in the DTU training data set is 512 * 640. So the shapes of depth_est, depth_gt and mask at stage0 should all be [8,64,80]. The shape of depth_gt and mask should not be [8,8,10]. I guess it is because you used the depth_gt that provided for MVSNet(size:128*160 for training). In my dtu_yao_1to8_inverse.py, we first read the depth_gt of the original size(1200 * 1600), and then resize and crop it to 512 * 640:

        h, w = hr_img.shape
        hr_img_ds = cv2.resize(hr_img, (w//2, h//2), interpolation=cv2.INTER_NEAREST)
        #crop
        h, w = hr_img_ds.shape
        target_h, target_w = 512, 640
        start_h, start_w = (h - target_h)//2, (w - target_w)//2
        hr_img_crop = hr_img_ds[start_h: start_h + target_h, start_w: start_w + target_w]

If the size of the depth_gt value you read is (128 * 160), it will be (64 * 80) after the above process, and then resized to (8 * 10) at stage0.

**Comment by LiPan123456789** [comment]
First of all, thank you for your reply.
I exported the shape of each stage of depth_gt and depth_est as follows:
![image](https://user-images.githubusercontent.com/97578468/202074406-106779bc-6a7b-4ab8-aea9-741d43442ff4.png)
Indeed it is as you said. But I am using the dtu_training dataset, how can I solve this problem, because in the readme there is only the download link for the evaluation set.

**Comment by bdwsq1996** [comment]
I just updated the readme, thank you for your reminder. You can download Depth_raw dataset from the link in readme now.

**Comment by lemonsstyle** [comment]
> I tested my code again, but I didn't find this problem. During the training, the size of the image in the DTU training data set is 512 * 640. So the shapes of depth_est, depth_gt and mask at stage0 should all be [8,64,80]. The shape of depth_gt and mask should not be [8,8,10]. I guess it is because you used the depth_gt that provided for MVSNet(size:128*160 for training). In my dtu_yao_1to8_inverse.py, we first read the depth_gt of the original size(1200 * 1600), and then resize and crop it to 512 * 640:
> 
> ```
>     h, w = hr_img.shape
>     hr_img_ds = cv2.resize(hr_img, (w//2, h//2), interpolation=cv2.INTER_NEAREST)
>     #crop
>     h, w = hr_img_ds.shape
>     target_h, target_w = 512, 640
>     start_h, start_w = (h - target_h)//2, (w - target_w)//2
>     hr_img_crop = hr_img_ds[start_h: start_h + target_h, start_w: start_w + target_w]
> ```
> 
> If the size of the depth_gt value you read is (128 * 160), it will be (64 * 80) after the above process, and then resized to (8 * 10) at stage0.

Hello, I also have this problem. So should I download the author's  ‘Depths’  file and replace the Depths folder in "dtu_training"?
